scripts/build/termux_step_get_dependencies.sh

```diff
@@ -1,79 +1,60 @@
-				continue
-			elif [[ "$PKG" == "ERROR" ]]; then
-				termux_error_exit "Obtaining buildorder failed"
-			fi
-			# llvm doesn't build if ndk-sysroot is installed:
-			if [[ "$PKG" == "ndk-sysroot" ]]; then continue; fi
-			read DEP_ARCH DEP_VERSION DEP_VERSION_PAC <<< $(termux_extract_dep_info $PKG "${PKG_DIR}")
-			local pkg_versioned="$PKG"
+			[[ "$PKG" == "ndk-sysroot" ]] && continue # llvm doesn't build if ndk-sysroot is installed:
+			read DEP_ARCH DEP_VERSION DEP_VERSION_PAC DEP_ON_DEVICE_NOT_SUPPORTED <<< $(termux_extract_dep_info $PKG "${PKG_DIR}")
+			local pkg_versioned="$PKG" build_dependency="false" force_build_dependency="$TERMUX_FORCE_BUILD_DEPENDENCIES"
 			[[ "${TERMUX_WITHOUT_DEPVERSION_BINDING}" == "false" ]] && pkg_versioned+="@$DEP_VERSION"
 			if [[ "$cyclic_dependence" == "false" ]]; then
 				[[ "$TERMUX_QUIET_BUILD" != "true" ]] && echo "Downloading dependency $pkg_versioned if necessary..."
-				local force_build_dependency="$TERMUX_FORCE_BUILD_DEPENDENCIES"
-				if [[ "$TERMUX_FORCE_BUILD_DEPENDENCIES" == "true" && "$TERMUX_ON_DEVICE_BUILD" == "true" ]] && ! termux_package__is_package_on_device_build_supported "$PKG_DIR"; then
+				if [[ "$TERMUX_FORCE_BUILD_DEPENDENCIES" == "true" && "$TERMUX_ON_DEVICE_BUILD" == "true" && "$DEP_ON_DEVICE_NOT_SUPPORTED" == "true" ]]; then
 					echo "Building dependency $PKG on device is not supported. It will be downloaded..."
 					force_build_dependency="false"
 				fi
 			else
-				local force_build_dependency="false"
+				force_build_dependency="false"
 			fi
-			local build_dependency="false"
 			if [[ "$force_build_dependency" = "true" ]]; then
+				termux_force_check_package_dependency && continue || :
 				[[ "$TERMUX_QUIET_BUILD" != "true" ]] && echo "Force building dependency $PKG instead of downloading due to -I flag..."
-				termux_force_check_package_dependency && continue
-				build_dependency=true
+				build_dependency="true"
 			else
 				if termux_package__is_package_version_built "$PKG" "$DEP_VERSION"; then
 					[[ "$TERMUX_QUIET_BUILD" != "true" ]] && echo "Skipping already built dependency $pkg_versioned"
 					continue
 				fi
 				if ! TERMUX_WITHOUT_DEPVERSION_BINDING="$([[ "${cyclic_dependence}" == "true" ]] && echo "true" || echo "${TERMUX_WITHOUT_DEPVERSION_BINDING}")" termux_download_deb_pac $PKG $DEP_ARCH $DEP_VERSION $DEP_VERSION_PAC; then
-					if [[ "$cyclic_dependence" == "true" || ( "$TERMUX_FORCE_BUILD_DEPENDENCIES" == "true" && "$TERMUX_ON_DEVICE_BUILD" == "true" ) ]]; then
-						echo "Download of $PKG$([[ "${TERMUX_WITHOUT_DEPVERSION_BINDING}" == "false" && "${cyclic_dependence}" == "false" ]] && echo "@$DEP_VERSION") from $TERMUX_REPO_URL failed"
-						return 1
-					else
-						echo "Download of $pkg_versioned from $TERMUX_REPO_URL failed, building instead"
-						build_dependency=true
-					fi
+					[[ "$cyclic_dependence" == "true" || ( "$TERMUX_FORCE_BUILD_DEPENDENCIES" == "true" && "$TERMUX_ON_DEVICE_BUILD" == "true" ) ]] \
```

Comment:
I think a bracegroup might be clearer here, even if it's a single command in the brace group.
```suggestion
					[[ "$cyclic_dependence" == "true" || ( "$TERMUX_FORCE_BUILD_DEPENDENCIES" == "true" && "$TERMUX_ON_DEVICE_BUILD" == "true" ) ]] && {
```

Reply:
Yep I see that now.